src/dataframerow/dataframerow.jl

```diff
@@ -129,8 +207,12 @@ Base.@propagate_inbounds Base.getindex(df::AbstractDataFrame, rowind::Integer, :
     DataFrameRow(df, rowind, :)
 Base.@propagate_inbounds Base.getindex(r::DataFrameRow, idx::ColumnIndex) =
     parent(r)[row(r), parentcols(index(r), idx)]
-Base.@propagate_inbounds Base.getindex(r::DataFrameRow, idxs::MultiColumnIndex) =
-    DataFrameRow(parent(r), row(r), parentcols(index(r), idxs))
+
+Base.@propagate_inbounds function Base.getindex(r::DataFrameRow, idxs::MultiColumnIndex)
+    dfr_tmp = DataFrameRow(parent(r), row(r), parentcols(index(r), idxs))
```

Comment:
Why do we have to call the constructor twice?

Reply:
This should be inlined optimized out by the compiler I think. The reason why we do it is that the first constructor is an outer constructor that recalculates the indices and properly creates `index(dfr_tmp)`.

I did not want to copy the logic here (I could) not to duplicate the code (which might be later easy to forget to fix) - but if you prefer I can do this.

The second call calls the inner constructor that "does nothing" (ie. it is only a wrapper of passed values - no checking of values takes place so this should be zero overhead and since `DataFrameRow` is immutable it should be fully optimized out)

Reply:
Maybe just add a comment explaining this.

Reply:
added